File: eio2023-ev/2023-12-10-ev/erinevused/solution/sol0.py
# naiivne O(N^2) keerukusega lahendus; jääb suuremates testides ajahätta

# erinevuse arvutamine täpselt ülesande püstituse järgi 
# alternatiivina võiks max(a[i],a[j])-min(a[i],a[j])
# asemel kirjutada ka abs(a[i]-a[j])

n = int(input())
a = input().split()
for i in range(n):
	a[i] = int(a[i])

s = 0
for i in range(n):
	for j in range(i):
		s += max(a[i], a[j]) - min(a[i], a[j])

print(s)

File: eio2023-ev/2023-12-10-ev/erinevused/solution/sol0.cpp
// naiivne O(N^2) keerukusega lahendus; jääb suuremates testides ajahätta

// erinevuse arvutamine täpselt ülesande püstituse järgi 
// alternatiivina võiks std::max(a[i],a[j])-std::min(a[i],a[j])
// asemel kirjutada ka std::abs(a[i]-a[j])

#include <iostream>
#include <vector>
#include <algorithm>

int main() {
	int n;
	std::cin >> n;
	std::vector<int> a(n);
	for (int i = 0; i < n; ++i) {
		std::cin >> a[i];
	}

	long long s = 0;
	for (int i = 0; i < n; ++i) {
		for (int j = 0; j < i; ++j) {
			s += std::max(a[i], a[j]) - std::min(a[i], a[j]);
		}
	}

	std::cout << s << '\n';
}
